nemo/core/neural_modules.py

```diff
@@ -392,6 +392,30 @@ def output_ports(self) -> Optional[Dict[str, NeuralType]]:
           A (dict) of module's output ports names to NeuralTypes mapping
         """
 
+    @property
```

Comment:
I like idea of moving the whole deployment to NeuralModule class, seems that having those methods here is the first step.

Having said that, I am not sure about the methods on them own, i.e. is disabling input/output ports for deployment is a _generic_ solution to all modules?

Besides, seems that you are changing the NM interface available to the user. Are you sure that we want to make them even "accessible"? In other words - I think those should be private methods of the class.

Reply:
I'm not sure without significant refactoring we can achieve making these fully private. We can change them to `_disabled_deployment_input_ports`, `_disabled_deployment_output_ports`, and `_prepare_for_deployment` so they're not part of the public NM API easily enough. Would that be sufficient for now since we intend to rework this further in the next release?

Reply:
I agree, makes sense for now.

My goal was to trigger the discussion, but it is clear that those changes shouldn't be part of the current release. Code freeze is not the right moment. That is why I am also quite surprised that during code freeze we change the public interface, understand what @okuchaiev is saying about "braking change" - but am still surprised... ;)